internal/rlimit.go

```diff
@@ -0,0 +1,54 @@
+package internal
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/cilium/ebpf/internal/unix"
+)
+
+func RemoveMemlockRlimit() error {
+	var oldLimit unix.Rlimit
+	if err := unix.Prlimit(0, unix.RLIMIT_MEMLOCK, nil, &oldLimit); err != nil {
+		return fmt.Errorf("retrieve RLIMIT_MEMLOCK: %s", err)
+	}
+
+	// Reduce the limit to zero. This is always allowed.
+	newLimit := unix.Rlimit{Cur: 0, Max: oldLimit.Max}
+	if err := unix.Prlimit(0, unix.RLIMIT_MEMLOCK, &newLimit, &oldLimit); err != nil {
+		return fmt.Errorf("lower RLIMIT_MEMLOCK: %s", err)
+	}
+
+	attr := BPFMapCreateAttr{
+		MapType:    2, /* Array */
+		KeySize:    4,
+		ValueSize:  4,
```

Comment:
- What about using a more targetted value like `oldLimit.Max` here instead of dropping the limit to zero? Creating a map that's highly likely to be over the current limit should be sufficient and less dangerous. If we go down this path, `getrlimit(2)` should be sufficient, no need to mutate system state.
- What about exposing this as a feature probe instead?

Reply:
Feature probe doesn't work since we need to change program global state.

Interesting idea re making a bigger map. The problem is that we might then run into memcg limits. When mcmg limit < memlock limit we would incorrectly deduce that the kernel uses memlimit accounting.

Reply:
Would there be a more reliable way to determine if memcg accounting is enabled? We could make a similarly wrong deduction if the memcg limit is set low enough.

Reply:
That's what I asked on the initial PR, and the answer was no unfortunately. If it turns out to be a problem we can remove the check and always bump the rlimit. This way round seems better for a start though.